src/backtrace.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @packageDocumentation
+ * Collects backtraces for executables for diagnostics
+ */
+import { stringifyError } from "./errors.js";
+import * as actionsCore from "@actions/core";
+import * as exec from "@actions/exec";
+import { readFile, readdir } from "node:fs/promises";
+import { promisify } from "node:util";
+import { gzip } from "node:zlib";
+import os from "os";
+
+export async function collectBacktraces(
+  prefixes: string[],
+): Promise<Map<string, string>> {
+  if (os.platform() === "darwin") {
```

Comment:
Better to use existing helpers for this: https://github.com/DeterminateSystems/detsys-ts/blob/main/src/actions-core-platform.ts#L140-L145